tests/login.spec.ts

```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env -S node --no-warnings --loader ts-node/esm
+// @ts-check
+
+import { test } from 'tstest'
+
+import { PuppetTest } from './fixtures/puppet-test/puppet-test.js'
+
+test('Integration testing', async t => {
+
+  const puppet = new PuppetTest()
+
+  puppet.on('login', _user => {
+    t.pass('should receive a login event')
```

Comment:
This test should be written better to reflect the system status change.

Consider change to:

```ts
const USER_ID = 'test-user'
const future = new Promise(resolve => puppet.on('login', resolve))
await puppet.start()
puppet.login(USER_ID)
await new Promise(resolve => setImmediate(resolve))

t.ok(future, USER_ID, `should get login event with ${USER_ID}`)
```

Reply:
You can't use ```t.ok``` like this. I changed it to ```t.same```